lib/msf/core/exploit/remote/http/nagios_xi/rce_check.rb

```diff
@@ -0,0 +1,72 @@
+# -*- coding: binary -*-
+
+module Msf::Exploit::Remote::HTTP::NagiosXi::RceCheck
+  # Uses the Nagios XI version to check which CVEs and related exploit modules the target is vulnerable to, if any
+  #
+  # @param version [Gem::Version] Nagios XI version
+  # @return [Hash, Msf::Exploit::CheckCode], Hash mapping CVE numbers to exploit module names if the target is vulnerable, Msf::Exploit::CheckCode otherwise
+  def nagios_xi_rce_check(version)
+    matching_exploits = {}
+    
+    # known exploits that affect versions prior to the one in the hash key
+    nagios_rce_version_prior = {
+      '5.6.6' => [['CVE-2019-15949', 'nagios_xi_plugins_check_ping_authenticated_rce']],
+      '5.8.0' => [['CVE-2020-35578', 'nagios_xi_plugins_filename_authenticated_rce']]
+    }
+
+    nagios_rce_version_prior.each do |fixed_version, info|
+      if version < Gem::Version.new(fixed_version)
+        matching_exploits = add_cve_module_to_hash(matching_exploits,info)
+      end
+    end
+
+    # known exploits that affect only the version in the hash key
+    nagios_rce_version_equals = {
+      '5.7.3' => [['CVE-2020-5791', 'nagios_xi_mibs_authenticated_rce'],['CVE-2020-5792', 'nagios_xi_snmptrap_authenticated_rce']]
+    }
```

Comment:
Using whitespace is cleaner and easier to edit for future additions:

```suggestion
    # known exploits that affect versions prior to the one in the hash key
    nagios_rce_version_prior = {
      '5.6.6' => [
        ['CVE-2019-15949', 'nagios_xi_plugins_check_ping_authenticated_rce']
      ],
      '5.8.0' => [
        ['CVE-2020-35578', 'nagios_xi_plugins_filename_authenticated_rce']
      ],
    }

    nagios_rce_version_prior.each do |fixed_version, info|
      if version < Gem::Version.new(fixed_version)
        matching_exploits = add_cve_module_to_hash(matching_exploits,info)
      end
    end

    # known exploits that affect only the version in the hash key
    nagios_rce_version_equals = {
      '5.7.3' => [
        ['CVE-2020-5791', 'nagios_xi_mibs_authenticated_rce'],
        ['CVE-2020-5792', 'nagios_xi_snmptrap_authenticated_rce'],
      ]
    }
```

Also, I'm not sure how you want to handle multiple CVES. Something to consider as some of the chained exploits use more than one CVE. The lazy option is to just pick one (usually the most significant). Using an array of CVEs is probably overkill.

However, there's also the issue of multiple modules for a single CVE, as you're using the CVE as an index in `matching_exploits`.